### DISCUSSION TITLE
How to add action to reload fzf with different input?

### DISCUSSION THREAD

**Discussion by kevinlam03** [ibhagwan/fzf-lua#1459] [Q&A] [answered]
I'm trying to add some functionality where my default <leader>f pulls up fzf.files and then I can switch to a different mode with actions (Ctrl-m to switch to git modified files). I want to keep all the defaults of the normal fzf.files call, but be able to switch to a different provider on some key map. Is there an easy way to achieve that? I tried the following and it works whenever I hit Ctrl-m, but now when I hit Enter it also opens up the thing I mapped to Ctrl-m (not sure why?).

in fzf lua config:
```
vim.keymap.set("n", "<leader>f", function()
    fzf.files({
        actions = {
            ["ctrl-m"] = function ()
                fzf.fzf_exec('git ls-files -m; git diff --name-only --staged', {
                    actions = {
                        ['default'] = fzf.actions.file_edit,
                    }
                })
            end,
            ["default"] = fzf.actions.file_edit,
        }
    })
end)
```

**Comment by ibhagwan** [comment]
> but now when I hit Enter it also opens up the thing I mapped to Ctrl-m (not sure why?).

Use `enter` in your actions table instead of `default`, I changed it a few months back to be more compatible with fzf logic.

**Reply by kevinlam03** [reply]
Hmm I already tried setting that as well. The following prints CTRL-M to :messages when I `<leader>f`, then `Enter` on the first entry.
```
vim.keymap.set("n", "<leader>f", function()
    fzf.files({
        actions = {
            ["ctrl-m"] = function ()
                vim.print("CTRL-M")
                fzf.fzf_exec('git ls-files -m; git diff --name-only --staged', {
                    actions = {
                        ['default'] = fzf.actions.file_edit,
                    }
                })
            end,
            ["enter"] = fzf.actions.file_edit,
        }
    })
end)
```

**Reply by ibhagwan** [reply, marked as answer]
The reason this happens is because `ctrl-m` is the same as enter (try it in the shell), similarly `ctrl-i` is tab, etc, switching to a different bind that doesn’t conflict will work, e.g. `ctrl-g`.

Btw, I would also add F1 help string (desc) and header hint:
```lua
vim.keymap.set("n", "<leader>gg", function()
  fzf.files({
    actions = {
      ["enter"] = fzf.actions.file_edit,
      ["ctrl-g"] = {
        desc = "switch-to-git",
        header = "Switch to git",
        fn = function()
          fzf.fzf_exec("git ls-files -m; git diff --name-only --staged", {
            actions = {
              ['enter'] = fzf.actions.file_edit,
            }
          })
        end,
      },
    }
  })
end)
```
